### PR TITLE
[Java/Spring] Support Spring Boot 3 and Jakarta EE 9

### PR DESCRIPTION
**Pull request by cachescrubber**
Support Spring Boot 3 and Jakarta EE 9

- https://github.com/springdoc/springdoc-openapi-demos/wiki/springdoc-openapi-2.x-migration-guide
- https://github.com/spring-projects/spring-boot/releases/tag/v3.0.0-M2

### PR checklist
 
- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by cachescrubber** [comment]
FYI @wing328 @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

It would need some small refinements, but the spring-boot library is already usable.

File: bin/configs/spring-boot-3.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+generatorName: spring
+outputDir: samples/openapi3/server/petstore/springboot-3
+inputSpec: modules/openapi-generator/src/test/resources/3_0/petstore.yaml
+templateDir: modules/openapi-generator/src/main/resources/JavaSpring
+additionalProperties:
+  groupId: org.openapitools.openapi3
+  documentationProvider: springdoc
+  artifactId: springboot
+  snapshotVersion: "true"
+  jakartaee: true
```
Comment (borsch) [inline bin/configs/spring-boot-3.yaml:10]: add this to `cliOptions` to update generator doc

Comment (cachescrubber) [reply bin/configs/spring-boot-3.yaml]: I
- introduced a documented cli option, **_useSpringBoot3_**.
- renamed _jakartaee_ to **_useJakartaEe_** (still used internally, but not exposed as an option).

WDYT?

**Comment by cachescrubber** [comment]
@wing328 `bin/configs/spring-boot-3.yaml ` requires java 17 to run in Ci. Could you please help to configure it?

File: bin/configs/spring-boot-3.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+generatorName: spring
+outputDir: samples/openapi3/server/petstore/springboot-3
+inputSpec: modules/openapi-generator/src/test/resources/3_0/petstore.yaml
+templateDir: modules/openapi-generator/src/main/resources/JavaSpring
+additionalProperties:
+  groupId: org.openapitools.openapi3
+  documentationProvider: springdoc
```
Comment (borsch) [inline bin/configs/spring-boot-3.yaml:7]: probably this should be defaulted to `springdoc` without ability to override it. same for `useJakartaEe=true`

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java [hunk 3]
```diff
@@ -204,6 +207,9 @@ public SpringCodegen() {
         cliOptions.add(CliOption.newBoolean(USE_SWAGGER_UI,
             "Open the OpenApi specification in swagger-ui. Will also import and configure needed dependencies",
             useSwaggerUI));
+        cliOptions.add(CliOption.newBoolean(USE_SPRING_BOOT3,
+            "Genrate code and provide dependencies for use with Spring Boot 3.x. (Use jakarta instead of javax in imports).",
```
Comment (borsch) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java:211]: typo `Genrate`

File: bin/configs/spring-boot-3.yaml (hunk 1 above)
Comment (borsch) [reply bin/configs/spring-boot-3.yaml]: @cachescrubber LGTM

**Review by borsch — APPROVED** [review]

File: bin/configs/spring-boot-3.yaml (hunk 1 above)
Comment (cachescrubber) [reply bin/configs/spring-boot-3.yaml]: Ok.

File: bin/configs/spring-boot-3.yaml (hunk 2 above)
Comment (cachescrubber) [reply bin/configs/spring-boot-3.yaml:7]: **springFoxDocumentationProvider** and swagger1AnnotationLibrary are not accepted with **useSpringBoot3**.

**Comment by wing328** [comment]
FYI. Merged c3a4914b224 to cover new folders in the spring tests (github workflow)

**Pull request merged**